cou/utils/openstack.py

```diff
@@ -13,125 +13,129 @@
-    ussuri: 17.1.6 focal: 17.2.3. In this case the identifier is the "minor" (17.1 and 17.2).
-    ussuri: 13.1.0 focal 13.1.1. In this case the identifier is the "micro" (13.1.0 and 13.1.1).
-
-    charm designate-bind workload_version tracks the version of the deb package bind9.
+    Charm designate-bind workload_version tracks the version of the deb package bind9 (rmadison).
+    For charm gnocchi it was used cmadison.
 
     [0] https://releases.openstack.org/
     [1] https://docs.openstack.org/charm-guide/latest/project/charm-delivery.html
 
-    :return: Ordered dictionary containing the version and the possible Openstack release.
+    :return: Ordered dictionary containing the version and the possible OpenStack release.
     :rtype: OrderedDict
     """
-    charms_types = {
-        "ceph": ["ceph-mon", "ceph-fs", "ceph-radosgw", "ceph-osd"],
-        "swift": ["swift-proxy", "swift-storage"],
-        "nova": ["nova-cloud-controller", "nova-compute"],
-        "ovn": ["ovn-dedicated-chassis", "ovn-central"],
-        "neutron": ["neutron-api", "neutron-gateway"],
-        "manila": ["manila-ganesha"],
-        "horizon": ["openstack-dashboard"],
-    }
-    with open("cou/utils/openstack_lookup.csv") as csv_file:
+    with open(CSV_FILE) as csv_file:
         openstack_lookup = OrderedDict()
-        service_column_index = 0
-        minor_column_index = 1
-        micro_column_index = 2
-        version_start_column_index = 3
         csv_reader = csv.reader(csv_file, delimiter=",")
         header = next(csv_reader)
         for row in csv_reader:
             service_dict: defaultdict[str, Any] = defaultdict(OrderedDict)
-            service = row[service_column_index]
-            minor = row[minor_column_index]
-            micro = row[micro_column_index]
-            for column_index in range(version_start_column_index, len(row), 2):
+            service = row[SERVICE_COLUMN_INDEX]
+            for column_index in range(VERSION_START_COLUMN_INDEX, len(row), 2):
                 os_version, _ = header[column_index].split("-")
                 min_version = row[column_index]
                 max_version = row[column_index + 1]
-                service_dict[os_version] = Versions(
+                service_dict[os_version] = VersionRange(
                     Version(min_version),
                     Version(max_version),
-                    True if minor == "TRUE" else False,
-                    True if micro == "TRUE" else False,
                 )
             openstack_lookup[service] = service_dict
-    for charm_type, charms in charms_types.items():
+    for charm_type, charms in CHARM_TYPES.items():
         for charm in charms:
             openstack_lookup[charm] = openstack_lookup[charm_type]
     return openstack_lookup
 
 
-OPENSTACK_LOOKUP = generate_openstack_lookup()
-
+class OpenStackLookup(object):
+    _OPENSTACK_LOOKUP: OrderedDict = OrderedDict()
```

Comment:
```suggestion
    _OPENSTACK_LOOKUP: Optional[OrderedDict] = None
```

Reply:
I tried to use like this before, but mypy will complain:
```
cou/utils/openstack.py:125: error: Item "None" of "OrderedDict[Any, Any] | None" has no attribute "get"  [union-attr]
cou/utils/openstack.py:135: error: Value of type "OrderedDict[Any, Any] | None" is not indexable  [index]
``` 
Line 125 : `if not self._OPENSTACK_LOOKUP.get(component)`
Line 135: `openstack_release, version_range in self._OPENSTACK_LOOKUP[component].items()`

Should I disable mypy on those lines? I didn't figure out a way to handle this